code/game/objects/items/broom.dm

```diff
@@ -0,0 +1,66 @@
+/obj/item/broom
+	name = "broom"
+	desc = "This is my BROOMSTICK! It can be used manually or braced with two hands to sweep items as you move. It has a telescopic handle for compact storage."
+	icon = 'icons/obj/janitor.dmi'
+	icon_state = "broom0"
+	lefthand_file = 'icons/mob/inhands/equipment/custodial_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/custodial_righthand.dmi'
+	force = 8
+	throwforce = 10
+	throw_speed = 3
+	throw_range = 7
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("swept", "brushed off", "bludgeoned", "whacked")
+	resistance_flags = FLAMMABLE
+	var/wielded = FALSE // track wielded status on item
+
+/obj/item/broom/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded=8, force_wielded=12, \
+				on_wield_callback=CALLBACK(src, .proc/on_wield), on_unwield_callback=CALLBACK(src, .proc/on_unwield))
+
+/// Callback triggered on wield of two handed item
+/obj/item/broom/proc/on_wield(mob/user)
+	wielded = TRUE
+	to_chat(user, "<span class='notice'>You brace the [src] against the ground in a firm sweeping stance.</span>")
+	RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/sweep)
+
+/// Callback triggered on unwield of two handed item
+/obj/item/broom/proc/on_unwield(mob/user)
+	wielded = FALSE
+	UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+
+/obj/item/broom/update_icon_state()
+	icon_state = "broom[wielded]"
```

Comment:
do this in the component

Reply:
Switched this from on item tracking to using the iconupdate callback